src/OpenTelemetry.Instrumentation.Http/TracerProviderBuilderExtensions.cs

```diff
@@ -30,13 +30,13 @@ public static class TracerProviderBuilderExtensions
         /// Enables HttpClient and HttpWebRequest instrumentation.
         /// </summary>
         /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
-        /// <param name="configureHttpClientInstrumentationOptions">HttpClient configuration options.</param>
         /// <param name="configureHttpWebRequestInstrumentationOptions">HttpWebRequest configuration options.</param>
+        /// <param name="configureHttpClientInstrumentationOptions">HttpClient configuration options.</param>
         /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
         public static TracerProviderBuilder AddHttpClientInstrumentation(
             this TracerProviderBuilder builder,
-            Action<HttpClientInstrumentationOptions> configureHttpClientInstrumentationOptions = null,
-            Action<HttpWebRequestInstrumentationOptions> configureHttpWebRequestInstrumentationOptions = null)
+            Action<HttpWebRequestInstrumentationOptions> configureHttpWebRequestInstrumentationOptions = null,
```

Comment:
Doesn't seem any less confusing or error-prone to me. IMO the solution is to not use "object" in the enrich callback signatures. Type safety/compile-time checking is our friend!

Reply:
@CodeBlanch I think the issue here is that many users call `AddHttpClientInstrumentation(options => {...})` on .NET Framework which is how it's called on .NET Core and the README also only has examples with this usage.

The problem is that this also compiles for .NET Framework which has different signature for `AddHttpClientInstrumentation` that take two arguments `HttpClientInstrumentationOptions` and `HttpWebRequestInstrumentationOptions` which are null by default. So when they setup `HttpWebRequestInstrumentationOptions` and call `AddHttpClientInstrumentation(options => {...})` it won't work as they should be calling `AddHttpClientInstrumentation(null, options => {...})`